src/wallet/wallet2.cpp

```diff
@@ -855,6 +855,46 @@ crypto::public_key wallet2::get_subaddress_spend_public_key(const cryptonote::su
   return D;
 }
 //----------------------------------------------------------------------------------------------------
+std::vector<crypto::public_key> wallet2::get_subaddress_spend_public_keys(uint32_t account, uint32_t begin, uint32_t end) const
```

Comment:
Hmm, isn't it possible to optimize even more, by not limiting the subaddress indices to be under the same major index?  I.e. the function signature would look like:

```c++
std::unordered_map<cryptonote::subaddress_index, crypto::public_key> wallet2::get_subaddress_spend_public_keys(const std::unordered_set<cryptonote::subaddress_index> &indices) const
```

Reply:
Yes, this is diminishing returns. "Unneeded" computation went down from 200*50 to 50. If you want to to a level above, you only gain not much, from 50 to 1. I don't think it's worth the bother, but if someone else does it I'd be OK with it.

Reply:
Ah, you're right. Not worth the bother.